docs/data/material/migration/migration-v5/migration-v5.md

```diff
@@ -14,12 +14,113 @@ In the `package.json` file, change the package version from `latest` to `next`.
+
+Apart from the support for Pigment CSS, the v6 release provides a stable API for adopting CSS variables which open more customization possibilities and offers a more enjoyable developer experience.
+
+Additionally, the v6 release does not include any breaking changes. Some APIs are deprecated to bring more consistent developer experience which are easily migrated using our codemod.
+
+## Supported browsers and Node versions
+
+The targets of the default bundle have changed in v6.
+
+The exact versions will be pinned on release from the browserslist query `"> 0.5%, last 2 versions, Firefox ESR, not dead, not IE 11, maintained node versions"`.
+
+<!-- #stable-snapshot -->
+
+<!-- Add  -->
+
+## Update React & TypeScript version
+
+<!-- Add minimum React & TypeScript version -->
+
+## New features and improvements
+
+### Container queries
+
+CSS Container Quries lets you apply styles based on the size of the container, rather than the viewport.
+
+The `theme` object now includes a field `containerQueries` which can be used to define container queries similar to `theme.breakpoints`.
+
+```js
+import { styled } from '@mui/material/styles';
+
+const MyComponent = styled('div')(({ theme }) => ({
+  [theme.containerQueries.up('sm')]: {
+    backgroundColor: 'red',
+  },
+}));
+```
+
+or using the `sx` prop:
+
+```js
+<Card sx={{
+  display: 'flex',
+  '@300px': {
+    flexDirection: 'column',
+  }
+}}>
+```
+
+<!-- TODO: add link to the docs -->
+
+To learn more about using container queries, check out the [Container Queries documentation](/system/container-queries/).
+
+## Stabilized APIs
+
+### CssVarsProvider and extendTheme
+
+CSS theme variables become a recommended way to customize the theme. The `CssVarsProvider` and `extendTheme` APIs are now stable.
+
+```diff
+- import { createTheme, ThemeProvider } from '@mui/material/styles';
++ import { extendTheme, CssVarsProvider } from '@mui/material/styles';
+
+- const theme = createTheme({...});
++ const theme = extendTheme({...});
+
+- <ThemeProvider theme={theme}>
++ <CssVarsProvider theme={theme}>
+```
+
+<!-- Link to CSS theme variables page -->
+
+To learn more about using CSS theme variables, check out the CSS theme variables page.
+
+### Grid
+
+The Grid v2 is now becoming the default Grid component. Check out the [migration guide](/material-ui/migration/migration-grid-v2/) to see the differences and improvements.
+
+## Deprecations
+
+### `components` and `componentsProps`
+
+<!-- TODO: add info and codemod instruction -->
+
+### Component style overrides
+
+<!-- TODO: add info and codemod instruction -->
+
+### System props
+
+System props are deprecated in `Box`, `Typography`, `Link`, `Grid`, and `Stack` components. Move all system props into the `sx` prop instead. Example:
+
+```diff
+- <Button mr={2}>...</Button>
++ <Button sx={{ mr: 2 }}>...</Button>
+```
+
+<!-- TODO: add codemod instruction -->
+
+## Pigment CSS integration
```

Comment:
```suggestion
## Pigment CSS integration (optional)
```

To make sure people know that they don't need to do this right away.

Reply:
Additional thing, we decided to keep in emotion the current overrides presendence (base styles vs variants), but change it in Pigment CSS. We need to add section on this and maybe troubleshooting guide if the styles differ.